### PR TITLE
Add qnn examples of clip models.

### PR DESCRIPTION
**Pull request by chinazhangchao**
## Describe your changes
Add qnn examples of clip models:

- openai/clip-vit-base-patch16
- openai/clip-vit-base-patch32
- openai/clip-vit-large-patch14

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [x] Make sure all tests can pass.
- [x] Update documents if necessary.
- [x] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [x] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

## Metrics
image

### CODE REVIEW

File: examples/clip/openai_clip-vit-base-patch32_ptq_qnn.json [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        }
+    },
+    "passes": {
+        "conversion": { "type": "OnnxConversion", "target_opset": 17 },
+        "qnn_preprocess": { "type": "QNNPreprocess" },
+        "quantization": {
+            "type": "OnnxStaticQuantization",
+            "quant_preprocess": true,
+            "data_config": "quant_data_config",
+            "op_types_to_quantize": [ "MatMul", "LayerNormalization", "Gemm", "Sigmoid", "Gelu" ],
```
Comment (chinazhangchao) [inline examples/clip/openai_clip-vit-base-patch32_ptq_qnn.json:115]: Note: the line `"op_types_to_quantize": [ "MatMul", "LayerNormalization", "Gemm", "Sigmoid", "Gelu" ],` in the json configurations are not effective due to known Olive issues, but has been left in place for future use.
https://github.com/microsoft/Olive/issues/1552

File: docs/source/examples.md [hunk 2]
```diff
@@ -16,6 +16,7 @@
 |Vision|stable diffusion <br> stable diffusion XL|[Link](https://github.com/microsoft/Olive/tree/main/examples/stable_diffusion)|`GPU`: with ONNX Runtime optimization for DirectML EP<br>`GPU`: with ONNX Runtime optimization for CUDA EP<br>`Intel CPU`: with OpenVINO toolkit
 ||squeezenet|[Link](https://github.com/microsoft/Olive/tree/main/examples/directml/squeezenet)|`GPU`: with ONNX Runtime optimizations with DirectML EP
 ||mobilenet|[Link](https://github.com/microsoft/Olive/tree/main/examples/mobilenet)|`Qualcomm NPU`: with ONNX Runtime static QDQ quantization for ONNX Runtime QNN EP
+||clip|[Link](https://github.com/microsoft/Olive/tree/main/examples/clip)|`Qualcomm NPU`: with ONNX Runtime static QDQ quantization for ONNX Runtime QNN EP
```
Comment (jambayk) [inline docs/source/examples.md:19]: The doc build test checks the link if it's present. Since the PR is not merged yet, this link doesn't exist. Please remove this change from this PR and submit a follow up PR once this gets merged.

Comment (jambayk) [reply docs/source/examples.md:19]: image

File: examples/clip/conda.yaml
Comment (jambayk) [inline examples/clip/conda.yaml:1]: I don't think this file is relevant?

File: examples/clip/conda_gpu.yaml
Comment (jambayk) [inline examples/clip/conda_gpu.yaml:1]: same

File: examples/clip/openai_clip-vit-base-patch16_ptq_qnn.json [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            "input_shapes": [ [ 10, 77 ], [ 1, 3, 224, 224 ], [ 10, 77 ] ],
+            "input_types": [ "int64", "float32", "int64" ],
+            "output_names": [ "logits_per_image" ],
+            "output_shapes": [ [ 1, 2 ] ]
+        }
+    },
+    "systems": {
+        "local_system": {
+            "type": "LocalSystem",
+            "accelerators": [ { "device": "npu", "execution_providers": [ "NPUExecutionProvider" ] } ]
```
Comment (jambayk) [inline examples/clip/openai_clip-vit-base-patch16_ptq_qnn.json:18]: QNNExecutionProvider

File: examples/clip/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# TODO(anyone): load_metrics was removed since 3.0.0. Using evaluate instead
+datasets<3.0.0
+evaluate
+scikit-learn==1.5.2
+transformers==4.44.2
```
Comment (jambayk) [inline examples/clip/requirements.txt:5]: does the transformers version limit apply to this example?

File: examples/clip/README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+Config file: [openai_clip-vit-base-patch32_ptq_qnn.json](openai_clip-vit-base-patch32_ptq_qnn.json)
+Config file: [openai_clip-vit-large-patch14_ptq_qnn.json](openai_clip-vit-large-patch14_ptq_qnn.json)
+
+**NOTE:** The model optimization part of the workflow can also be done on a Linux/Windows machine with a different onnxruntime package installed. Remove the `"evaluators"` and `"evaluator"` sections from the configuration file to skip the evaluation step.
+
+## How to run
+### Pip requirements
+Install the necessary python packages:
+```sh
+# [CPU]
```
Comment (jambayk) [inline examples/clip/README.md:27]: cpu/gpu installations are not relevant here

File: examples/clip/openai_clip-vit-base-patch16_ptq_qnn.json [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            "metrics": [
+                {
+                    "name": "accuracy",
+                    "type": "accuracy",
+                    "backend": "huggingface_metrics",
+                    "data_config": "metric_data_config",
+                    "sub_types": [
+                        { "name": "accuracy", "priority": 1, "goal": { "type": "max-degradation", "value": 0.05 } }
+                    ],
+                    "user_config": {
```
Comment (jambayk) [inline examples/clip/openai_clip-vit-base-patch16_ptq_qnn.json:59]: nit: since the target ep is qnn, this config is not needed if you want to run on qnn ep.

**Review by jambayk — APPROVED** [review]
